### PR TITLE
Fix CJK line Breaking

### PR DESCRIPTION
**Pull request by Fahad-Alsaidi**
This mimic the old stlye in line breaking though we should find a better solution for line breaking in general.

### CODE REVIEW

File: scribus/pageitem_textframe.cpp [hunk 1]
```diff
@@ -2178,6 +2178,10 @@ void PageItem_TextFrame::layout()
 			// remember possible break
 			if (i + 1 < glyphRuns.count() && glyphRuns[i + 1].hasFlag(ScLayout_LineBoundry))
 			{
+				if((itemText.text(currentIndex -1) != SpecialChars::CJK_NOBREAK_AFTER) &&
+									   (itemText.text(currentIndex) != SpecialChars::CJK_NOBREAK_BEFORE))
+					current.rememberBreak(i-1, breakPos, style.rightMargin());
+
```
Comment (khaledhosny) [inline scribus/pageitem_textframe.cpp]: `currentIndex` is the glyph cluster index not story text index, you should use `current.glyphs[currentIndex - 1].lastChar()` and `current.glyphs[currentIndex].firstChar()`. and make sure `i` is not zero.

**Comment by Fahad-Alsaidi** [comment]
updated!

File: scribus/pageitem_textframe.cpp [hunk 2]
```diff
@@ -2178,6 +2178,12 @@ void PageItem_TextFrame::layout()
 			// remember possible break
 			if (i + 1 < glyphRuns.count() && glyphRuns[i + 1].hasFlag(ScLayout_LineBoundry))
 			{
+
+				if( current.glyphs.length() > 1 && (current.glyphs[currentIndex -1].lastChar() != SpecialChars::CJK_NOBREAK_AFTER) &&
```
Comment (khaledhosny) [inline scribus/pageitem_textframe.cpp]: The space should be before the opening bracket, also there should be spaces around both sides of the minus operator.

**Comment by Fahad-Alsaidi** [comment]
updated!

**Pull request merged**